src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -182,6 +182,12 @@ public void OnResponseData(ReadOnlySpan<byte> buffer, bool endStream)
 
                     Debug.Assert(!_responseComplete);
 
+                    if (_responseBuffer.ActiveSpan.Length + buffer.Length > InitialWindowSize)
```

Comment:
I assume there's no way this would overflow?

Reply:
Not as the code currently works; the response buffer is limited to InitialWindowSize (64K) and the newly received buffer can't be larger than a single frame, i.e. 16K.